packages/amplify-cli/src/execution-manager.ts

```diff
@@ -39,6 +39,33 @@ export function isContainersEnabled(context) {
 }
 
 async function selectPluginForExecution(context: Context, pluginCandidates: PluginInfo[]): Promise<PluginInfo> {
+  const pluginCandidatesCategorySet = new Set<string>();
+  const overidedcategories = ['auth'];
+
+  pluginCandidates.forEach(plugin => {
+    pluginCandidatesCategorySet.add(plugin.manifest.name);
+  });
+
+  // overrided packageName format : @aws-amplify/amplify-category-<catgoryName>
+  const pluginName = pluginCandidatesCategorySet.values().next().value;
+  if (pluginCandidatesCategorySet.size == 1 && overidedcategories.includes(pluginName)) {
```

Comment:
what happens in the case of hosting - where we have two category plugins and when we introduce this new plugin with overrides..what options will the CLI show then?

Reply:
I am first checking if the overrided package support aboveName and feature flag is on for the category.

if we didn't find the overrides package , then we check if the plugin size is 2 , then we have {overidedPackage , nonOverided package}

In case of hosting , if the overrides for hosting is there , it will get selected ,otherwise this will prompt for selection.

Reply:
so it will still show 2 option or 3?

Reply:
3 options for hosting